test/gurka/test_shortcut.cc

```diff
@@ -289,3 +289,81 @@ TEST(Shortcuts, ShortcutsInBins) {
 
   EXPECT_EQ(shortcut_cnt, 1);
 }
+
+TEST(Shortcuts, ShortcutRestrictions) {
+  const std::string ascii_map = R"(
+  A--B--C--D--E--F
+
+  G--H--I--J--K--L
+  )";
+
+  // the first line should produce only one shortcut, the second two
+  gurka::ways ways = {
+      {"AB", {{"highway", "motorway"}, {"name", "highway"}}},
+      {"BC", {{"highway", "motorway"}, {"name", "highway"}}},
+      {"CD", {{"highway", "motorway"}, {"name", "highway"}, {"maxweight", "3.5"}}},
+      {"DE", {{"highway", "motorway"}, {"name", "highway"}, {"maxweight", "8"}}},
```

Comment:
amended the test so it'll at least once get to the "return false" to prove we're breaking here too